featurebyte/query_graph/graph.py

```diff
@@ -281,6 +287,27 @@ def prune(self, target_node: Node, aggressive: bool) -> GraphNodeNameMap:
         )
         return pruned_graph, node_name_map
 
+    def crop(self, target_node_names: List[str]) -> GraphCroppingOutput:
+        """
+        Crop the query graph and return the cropped graph & mapped node.
+
+        To crop the graph, this function first traverses from the target node to the input node.
+        The unused branches of the graph will get pruned in this step. After that, a new graph is
+        reconstructed by adding the required nodes back. Crop is less expensive than prune as it
+        does not extract the operation structure.
```

Comment:
From caller's perspective, is cropping a graph equivalent to non-aggressive pruning for a list of target node (aside from whether the operation structure is extracted internally)?

When doing non-aggressive pruning, would it make sense to use crop if it is less expensive?

Reply:
To make it clearer, i am thinking to change all the non-aggressive pruning from the client side to `crop` and `prune` should only have aggressive mode (aggressive mode parameter will be removed). Note that the amount of changes could be quite big if we make it in the same PR.

Reply:
Got it, that makes sense to me.

Another minor comment is that I find the name `crop` unclear in conveying what the operation means and when to use it. Wondering if something like `quick_prune()` where input parameter is a list of nodes is clearer: it is non-aggressive pruning but faster and can work on a list of target nodes, and `prune()` is expected to be slower because it prunes more aggressively.